Source/WebCore/Modules/geolocation/GeoNotifier.cpp

```diff
@@ -99,10 +99,6 @@ void GeoNotifier::timerFired()
 {
     m_timer.stop();
 
-    // Protect this GeoNotifier object, since it
-    // could be deleted by a call to clearWatch in a callback.
-    Ref<GeoNotifier> protectedThis(*this);
```

Comment:
It seems a bit risky to remove these in the same PR. Maybe we can do it in a separate PR?

Reply:
I have validated that these functions are all called from timers, it's actually very easy to validate. And what's a best way to validate my Timer change is doing the right thing?

Reply:
How strongly do you feel about splitting the PR? My preference would be to land in a single PR as I don't think it is particularly risky in this case. But I'll split if you insist :)

Reply:
I think it’s great to have them in the same pull request. To me a risk here is that if the `ref` function detection metaprogramming gets screwed up then the code still **looks** safe but has become unsafe. It’s not so common for us to have that sort of subtle dependency. I wonder if there’s a way to make this obvious. For example, would the safer C++ static analysis detect that situation? If so, then I am not worried.

Reply:
I can't speak to the static analysis but my plan is actually to require classes using Timers to either be ref-counted or CanMakeCheckedPtr and have the Timer take a Ref / CheckedRef before calling the member function. Then we can drop the unsafe constructor. During the transition period, we can add an exception list (like we did for `IsDeprecatedWeakRefSmartPointerException`) so that new code has to be made safe (while we port old code).

Reply:
Landing in a single PR sounds okay to me.